src/Builder/ClassModifier/ClassInstanceClassModifier.php

```diff
@@ -77,16 +85,23 @@ public function wireClassInstance(
         }
 
         if ($this->classMethodChecker->isMethodReturnArray($methodNode)) {
+            $indexAddUseVisitor = $this->createIndexAddUseVisitor($index);
+
             $visitors = [
                 new AddUseVisitor($classNameToAdd),
                 new AddPluginToPluginListVisitor(
                     $targetMethodName,
                     $classNameToAdd,
                     $before,
                     $after,
+                    $this->createIndexExpr($index),
                 ),
             ];
 
+            if ($indexAddUseVisitor) {
+                $visitors[] = $indexAddUseVisitor;
+            }
```

Comment:
```suggestion
            if ($this->isIndexClassMethodLike($index)) {
                $visitors[] = new AddUseVisitor($this->getClassNameFromIndex());
            }
```

IMO it would make the code a bit more readable.